lib/commands/app/purge.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright 2016-2020 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { flags } from '@oclif/command';
+import Command from '../../command';
+import * as cf from '../../utils/common-flags';
+import { getBalenaSdk, stripIndent } from '../../utils/lazy';
+import { tryAsInteger } from '../../utils/validation';
+
+interface FlagsDef {
+	help: void;
+}
+
+interface ArgsDef {
+	name: string;
+}
+
+export default class AppRestartCmd extends Command {
+	public static description = stripIndent`
+		Purge data from an application.
+
+		Purge data from all devices belonging to an application.
+		This will clear the application's /data directory.
+`;
+	public static examples = ['$ balena app purge MyApp'];
+
+	public static args = [
+		{
+			name: 'name',
+			description: 'application name or numeric ID',
+			required: true,
+		},
+	];
+
+	public static usage = 'app purge <name>';
+
+	public static flags: flags.Input<FlagsDef> = {
+		help: cf.help,
+	};
+
+	public static authenticated = true;
+
+	public async run() {
+		const { args: params } = this.parse<FlagsDef, ArgsDef>(AppRestartCmd);
+
+		const balena = getBalenaSdk();
+
+		// balena.models.application.purge only accepts a numeric id
+		// so we must first fetch the app to get it's id, if we have been given a name
+		let nameOrId = tryAsInteger(params.name);
```

Comment:
There is this issue: https://github.com/balena-io/balena-cli/issues/1842

With pre-existing commands, we have been using `tryAsInteger` as a way of maintaining backwards compatibility. But this PR introduces a new command, so it is an opportunity of "doing it right", starting by deciding what this means. :-)

By the way, I believe the API recently introduced app UUIDs:
https://github.com/balena-io/open-balena-api/pull/491

Not sure if app UUIDs are in production yet, but if they are, this PR is also an opportunity of introducing their usage in the CLI.

Reply:
> But this PR introduces a new command, so it is an opportunity of "doing it right", starting by deciding what this means. :-)

(just copying what I said in FD) I agree that we need to address that, but this command doesn't introduce a new problem (we already have it in multiple commands) so I think it should be merged.  I think we should then have a wider discussion about how to address this at the framework level (what conventions to adopt), and get consensus from @shaunmulligan and others.

> Not sure if app UUIDs are in production yet, but if they are, this PR is also an opportunity of introducing their usage in the CLI.

There's no mention of app UUIDs in the SDK documention, so I don't think so.

BTW - we are having to 'manually' support appNames in this command, because purge only supports IDs - not sure what the situation will be with app UUIDs, but I hope that they will be implemented across the board in the SDK (for the sake of autogeneration of the CLI).